lib/HLSL/DxilConvergent.cpp

```diff
@@ -156,11 +156,12 @@ bool DxilConvergentMark::PropagateConvergentImpl(Value *V, Function *F,
       // TODO: static indexing cbuffer is fine.
       return true;
     }
-  } else {
+  } else if (!isa<Argument>(V)) {
```

Comment:
What will happen if argument also used directly as sample coordinate?

Texture2D tex0 : register(t0);
SamplerState samp0 : register(s0);

 float4 main(float2 u : TEXCOORD0, float v : TEXCOORD1) : SV_TARGET {
  float4 result = float4(0,0,0,0);
  int i = 0;
  [branch]
  if (u > 0.5) {
    float2 coord = float2(
      EvaluateAttributeAtSample(u.x, i),
      EvaluateAttributeAtSample(v, i)
    );
    result += tex0.Sample(samp0, u);
    result += tex0.Sample(samp0, coord);
  }
  return result;
}

Reply:
Turns out this doesn't work, loadInput gets sank. I think I'll do your suggested high level intrinsics solution.